### ISSUE TITLE
Notifications components update

### ISSUE THREAD

**Issue by Memo-Es** [opencollective/opencollective#4681]
We are bringing together the different notifications components we have in the platform:
* Top banner
* Message box
* Toast notification

![❖  Notification Components](https://user-images.githubusercontent.com/49539954/133368152-8efb11aa-062f-47e3-853f-460ebc53ab04.png)

Solving #4646 and #4613 

CC @Betree

**Comment by Betree** [comment]
Great, thank you @Memo-Es! That will definitely resolve https://github.com/opencollective/opencollective/issues/4646.

To simplify the review and implementation, we should add stories for these components on https://styleguide.opencollective.com.

**Comment by Betree** [comment]
@Memo-Es I mistakingly closed this one instead of https://github.com/opencollective/opencollective/issues/4646, fixing that now. Thanks for noticing!

**Comment by gustavlrsn** [comment]
What behavior do we want for dismissing notification banners? We could adapt and utilize the `DismissibleMessage` component to persistently dismiss notification banners, if we give the notification some unique ID.

**Comment by gustavlrsn** [comment]
I've added updated notification banners to the styleguide in the PR linked above: https://opencollective-styleguide-git-feat-notifi-3d68a0-opencollective.vercel.app/?path=/story/design-system-notificationbar--default-story
